src/shogun/kernel/GaussianARDKernel.cpp

```diff
@@ -124,7 +124,7 @@ float64_t CGaussianARDKernel::get_parameter_gradient_helper(
 
 	if (!strcmp(param->m_name, "weights"))
 	{
-		linalg::add(avec, bvec, bvec, 1.0, -1.0);
+		bvec=linalg::add(avec, bvec, 1.0, -1.0);
```

Comment:
@karlnapf yeah but presently these operations work the same way for matrices as well (in-place not allowed if we're not using an explicit backend). Maybe I'll think of something about how to handle this backend-independently.